wandb/sdk/wandb_init.py

```diff
@@ -1281,15 +1281,32 @@ def try_create_root_dir(settings: Settings) -> None:
             if the parent directory is not writable.
     """
     try:
-        if not os.path.exists(settings.root_dir):
-            os.makedirs(settings.root_dir, exist_ok=True)
+        root_dir = settings.root_dir
+
+        if os.path.exists(root_dir):
+            if not os.access(root_dir, os.R_OK):
+                root_dir = tempfile.gettempdir()
+                wandb.termwarn(
+                    f"Path {settings.root_dir} wasn't readable, using system temp directory {root_dir}.",
+                    repeat=False,
+                )
+            elif not os.access(root_dir, os.W_OK):
+                root_dir = tempfile.gettempdir()
+                wandb.termwarn(
+                    f"Path {settings.root_dir} wasn't writable, using system temp directory {root_dir}.",
+                    repeat=False,
+                )
+
+        settings.root_dir = root_dir
+        os.makedirs(settings.root_dir, exist_ok=True)
     except OSError:
-        temp_dir = tempfile.gettempdir()
+        root_dir = tempfile.gettempdir()
         wandb.termwarn(
-            f"Path {settings.root_dir} wasn't writable, using system temp directory {temp_dir}.",
+            f"Unable to create root directory {settings.root_dir}, using system temp directory {root_dir}.",
```

Comment:
```suggestion
            (
                f"Unable to create root directory {settings.root_dir},"
                f" using system temp directory {root_dir}."
            ),
```

If ruff-format reconcatenates these, fine to keep as is (I think we have it set to 100 characters, but this looks like it might be more than that)